src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptionConverters.java

```diff
@@ -232,12 +232,19 @@ private static Label convertOptionsLabel(String input, @Nullable Object conversi
         return Label.parseCanonical(input);
       }
       Preconditions.checkArgument(
-          conversionContext instanceof RepositoryMapping,
+          conversionContext instanceof RepositoryMapping
```

Comment:
quick question -- this change shouldn't be necessary, right? everything in the platform mapping file should be anchored to the main repo? (it looks like you're passing around a `mainRepoContext` that always has `RepositoryName.MAIN`, instead of a `mainRepoMapping`; I think the latter would be less cognitive burden.)

Reply:
I found it confusing that `RepoContext` is not a valid `conversionContext`, but `RepositoryMapping` is - it doesn't matter that much though and `RepoContext` does have redundant information, so I reverted this part.